MS Linker may fail on 32 bit debug builds: stack overflow or LNK1318 fatal error.

Given this D code
```d
struct S
{
    void*[300_000] arr;
}

void main()
{
}
```
compiled with `dmd -g -m32 app.d` may fail depending on the linker version.

With Microsoft ® Incremental Linker Version 14.44.35214.0, which is distributed with Visual Studio Community 2022 17.14.9 and typically installed in `C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.44.35207\bin\HostX64\x86\link.exe`, fails with either exit status -1073740791 (stack buffer overrun) or `fatal error LNK1318: Unexpected PDB error; LIMIT (12) ''`.

The older version 14.34.31937.0, distributed with  Visual Studio Community 2022 17.4.4, installation path `...\MSVC\14.34.31933\...`, does not have this problem.

I have reported this with Microsoft, https://developercommunity.visualstudio.com/t/Linker-fails-with-STATUS_STACK_BUFFER_OV/10942393

We are not affected by this issue anymore as it was easy to remove this kind of code from our code base, once we found the culprit. Reported here for reference, and perhaps there is a better way to generate debugging information on 32 bit builds.

--- a/compiler/src/dmd/backend/mscoffobj.d
+++ b/compiler/src/dmd/backend/mscoffobj.d
@@ -460,6 +460,9 @@ private void syment_set_name(SymbolTable32* sym, const(char)* name)
     size_t len = strlen(name);
     if (len > 8)
     {   // Use offset into string table
+        // symbols larger than 64kB crash link.exe 14.44 or later
+        if (len > CV8_MAX_SYMBOL_LENGTH)
+            len = CV8_MAX_SYMBOL_LENGTH;
         IDXSTR idx = MsCoffObj_addstr(string_table, name[0 .. len]);
         sym.Zeros = 0;
         sym.Offset = idx;

--- a/compiler/test/runnable/test21757.d
+++ b/compiler/test/runnable/test21757.d
@@ -1,0 +1,10 @@
+// https://github.com/dlang/dmd/issues/21757
+
+struct S
+{
+    void*[300_000] arr; // generates stupidly large symbols for RTImfoImpl!() that crash MS link.exe
+}
+
+void main()
+{
+}

